src/Lykke.AlgoStore.Service.Statistics.Services/StatisticsService.cs

```diff
@@ -0,0 +1,106 @@
+            }
+
+            var assetPairResponse = await _assetsService.TryGetAssetPairAsync(algoInstance.AssetPairId);
+            ValidateAssetPair(algoInstance.AssetPairId, assetPairResponse);
+
+            var tradedAsset = await _assetsService.TryGetAssetAsync(algoInstance.IsStraight
+                ? assetPairResponse.BaseAssetId
+                : assetPairResponse.QuotingAssetId);
+            ValidateAssetResponse(tradedAsset);
+
+            if (algoInstance.AlgoInstanceType == CSharp.AlgoTemplate.Models.Enumerators.AlgoInstanceType.Live)
+            {
+                var walletBalances = await _walletBalanceService.GetWalletBalancesAsync(algoInstance.WalletId, assetPairResponse);
+                var clientBalanceResponseModels = walletBalances.ToList();
+                var latestWalletBalance = await _walletBalanceService.GetTotalWalletBalanceInBaseAssetAsync(
+                    algoInstance.WalletId, statisticsSummary.UserCurrencyBaseAssetId, assetPairResponse);
+
+                statisticsSummary.LastTradedAssetBalance = (double) (clientBalanceResponseModels.FirstOrDefault(b => b.AssetId == tradedAsset.Id)?.Balance ?? 0);
+                statisticsSummary.LastAssetTwoBalance = (double) (clientBalanceResponseModels.FirstOrDefault(b => b.AssetId != tradedAsset.Id)?.Balance ?? 0);
+                statisticsSummary.LastWalletBalance = latestWalletBalance;
+            }
+
+            statisticsSummary.NetProfit = statisticsSummary.InitialWalletBalance.Equals(0.0) ? 0 : Math.Round(
+                ((statisticsSummary.LastWalletBalance - statisticsSummary.InitialWalletBalance) /
+                 statisticsSummary.InitialWalletBalance) * 100, 2, MidpointRounding.AwayFromZero);
+
+            await _statisticsRepository.CreateOrUpdateSummaryAsync(statisticsSummary);
+
+            return statisticsSummary;
+        }
+
+        private static void ValidateAssetResponse(Asset assetResponse)
+        {
+            if (assetResponse == null)
+                throw new ValidationException("Asset was not found");
+        }
+
+        private static void ValidateAssetPair(string assetPairId, AssetPair assetPair)
+        {
+            if (assetPair == null)
+                throw new ValidationException($"AssetPair: {assetPairId} was not found");
+
+            if (assetPair.IsDisabled)
+                throw new ValidationException($"AssetPair {assetPairId} is temporarily disabled");
+        }
+
+        private static void ValidateInstanceId(string instanceId)
+        {
+            if (string.IsNullOrEmpty(instanceId))
+                throw new ValidationException(Phrases.InstanceIdCannotBeEmpty);
+        }
+
+        private static void ValidateClientId(string instanceId)
+        {
+            if (string.IsNullOrEmpty(instanceId))
+                throw new ValidationException(Phrases.ClientIdCannotBeEmpty);
+        }
+    }
+}
```

Comment:
Why not move the validation logic in different class to support higher cohesion ?

Reply:
This will be done in future with another task, when we move ALL other statistics related stuff into statistics service

Reply:
Ok